### PR TITLE
transfer launch params (env and envFIle) to packager

### PR DESCRIPTION
**Pull request by JiglioNero**
#1235 add runOption setter to packager for transfer launch params
added fallback to `projectRoot/.env` file if no `env` or `envFile` are provided

### CODE REVIEW

File: src/common/packager.ts [hunk 1]
```diff
@@ -145,6 +157,7 @@ export class Packager {
                 // The packager will continue running while we debug the application, so we can"t
                 // wait for this command to finish
 
+
```
Comment (RedMickey) [inline src/common/packager.ts:160]:
```suggestion
```

File: test/extension/generalMobilePlatform.test.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for details.
 
+
```
Comment (RedMickey) [inline test/extension/generalMobilePlatform.test.ts:4]:
```suggestion
```

File: test/extension/generalMobilePlatform.test.ts [hunk 3]
```diff
@@ -29,5 +30,38 @@ suite("generalMobilePlatform", function () {
...
+            test("args from envFile should not overwrite existing variables", function() {
+                assert.deepEqual(GeneralMobilePlatform.getEnvArgument(origEnv, null, envFile), {
+                    "test1": "origEnv",
+                     "test2": "origEnv",
+                      "test3": "origEnv",
+                       "test4": "envFile",
+                        "test5": "envFile"});
+            });
+
+            test("args from envFile and original args should be overwrited by env args", function() {
```
Comment (SounD120) [inline test/extension/generalMobilePlatform.test.ts:57]: I think it makes sense to add one another test for case if `envFile` is null or undefined.

File: test/extension/generalMobilePlatform.test.ts [hunk 4]
```diff
@@ -29,5 +30,38 @@ suite("generalMobilePlatform", function () {
                 assert.equal(GeneralMobilePlatform.getOptFromRunArgs(args, "param4", false), undefined);
             });
         });
+
+        suite("getEnvArgument", function() {
+            let origEnv: any = {"test1": "origEnv", "test2": "origEnv", "test3": "origEnv"};
+            let env: any = {"test2": "env", "test3": "env", "test4": "env"};
+            let envForFile: string = "test3=envFile\ntest4=envFile\ntest5=envFile";
+            let envFile: string = ".env";
```
Comment (SounD120) [inline test/extension/generalMobilePlatform.test.ts:38]: Make sure to create this file not in the repository root, but in special temporary resources folder to avoid any side effects for this test.

File: src/common/packager.ts [hunk 5]
```diff
@@ -45,26 +47,31 @@ export class Packager {
     private static OPN_PACKAGE_MAIN_FILENAME = "index.js";
     private static fs: FileSystem = new Node.FileSystem();
     private expoHelper: ExponentHelper;
+    private runOptions: IRunOptions;
```
Comment (SounD120) [inline src/common/packager.ts:50]: Where in the code we assign this variable?

File: test/extension/generalMobilePlatform.test.ts [hunk 6]
```diff
@@ -29,5 +31,47 @@ suite("generalMobilePlatform", function () {
...
             });
         });
+
+        suite("getEnvArgument", function() {
+            let origEnv: any = {"test1": "origEnv", "test2": "origEnv", "test3": "origEnv"};
+
+            let env: any = {"test2": "env", "test3": "env", "test4": "env"};
+
+            let envForFile: string = "test3=envFile\ntest4=envFile\ntest5=envFile";
+            let envFile: string = path.join(__dirname, "../resources/auxiliaryFiles/.env");
```
Comment (SounD120) [inline test/extension/generalMobilePlatform.test.ts:41]:
```suggestion
            const envFile: string = path.join(__dirname, "..", "resources", "auxiliaryFiles", ".env");
```

File: test/extension/generalMobilePlatform.test.ts [hunk 7]
```diff
@@ -29,5 +31,47 @@ suite("generalMobilePlatform", function () {
                 assert.equal(GeneralMobilePlatform.getOptFromRunArgs(args, "param4", false), undefined);
             });
         });
+
+        suite("getEnvArgument", function() {
+            let origEnv: any = {"test1": "origEnv", "test2": "origEnv", "test3": "origEnv"};
+
+            let env: any = {"test2": "env", "test3": "env", "test4": "env"};
+
+            let envForFile: string = "test3=envFile\ntest4=envFile\ntest5=envFile";
```
Comment (SounD120) [inline test/extension/generalMobilePlatform.test.ts:40]:
```suggestion
            const envForFile: string = "test3=envFile\ntest4=envFile\ntest5=envFile";
```

File: test/extension/generalMobilePlatform.test.ts [hunk 8]
```diff
@@ -29,5 +31,47 @@ suite("generalMobilePlatform", function () {
                 assert.equal(GeneralMobilePlatform.getOptFromRunArgs(args, "param4", false), undefined);
             });
         });
+
+        suite("getEnvArgument", function() {
+            let origEnv: any = {"test1": "origEnv", "test2": "origEnv", "test3": "origEnv"};
+
+            let env: any = {"test2": "env", "test3": "env", "test4": "env"};
```
Comment (SounD120) [inline test/extension/generalMobilePlatform.test.ts:38]:
```suggestion
            const env: any = {"test2": "env", "test3": "env", "test4": "env"};
```

File: test/extension/generalMobilePlatform.test.ts [hunk 9]
```diff
@@ -29,5 +31,47 @@ suite("generalMobilePlatform", function () {
                 assert.equal(GeneralMobilePlatform.getOptFromRunArgs(args, "param4", false), undefined);
             });
         });
+
+        suite("getEnvArgument", function() {
+            let origEnv: any = {"test1": "origEnv", "test2": "origEnv", "test3": "origEnv"};
```
Comment (SounD120) [inline test/extension/generalMobilePlatform.test.ts:36]:
```suggestion
            const origEnv: any = {"test1": "origEnv", "test2": "origEnv", "test3": "origEnv"};
```

File: test/extension/generalMobilePlatform.test.ts [hunk 10]
```diff
@@ -29,5 +31,47 @@ suite("generalMobilePlatform", function () {
...
+            test("args from envFile should not overwrite existing variables", function() {
+                assert.deepEqual(GeneralMobilePlatform.getEnvArgument(origEnv, null, envFile), {
+                    "test1": "origEnv",
+                     "test2": "origEnv",
+                      "test3": "origEnv",
+                       "test4": "envFile",
+                        "test5": "envFile"});
+            });
+
+            test("args from envFile and original args should be overwrited by env args", function() {
```
Comment (RedMickey) [inline test/extension/generalMobilePlatform.test.ts:67]:
```suggestion
            test("args from envFile and original args should be overwritten by env args", function() {
```

File: test/extension/generalMobilePlatform.test.ts [hunk 11]
```diff
@@ -29,5 +31,47 @@ suite("generalMobilePlatform", function () {
...
+
+            setup(() => {
+                fs.writeFileSync(envFile, envForFile);
+            });
+
+            teardown(() => {
+                fs.unlinkSync(envFile);
+            });
+
+            test("existing args should not depends from null or undefined env and envFile", function() {
```
Comment (RedMickey) [inline test/extension/generalMobilePlatform.test.ts:51]:
```suggestion
            test("existing args should not depend on null or undefined env and envFile", function() {
```

File: README.md [hunk 12]
```diff
@@ -404,6 +404,13 @@ If you use Android, you need to change the debug server by:
 
 **NOTE:** Some aspects of React Native hard-code the port to the default as specified in [this issue](https://github.com/facebook/react-native/issues/9145).
 
+Extension supports passing custom enviroment variables for the `react-native` packager process context. To add custom variables you can create .env file in the root folder of your project and add needed environment variables in the following format:
```
Comment (SounD120) [inline README.md:407]:
```suggestion
### Custom environment variables
Extension supports passing custom environment variables for the React Native Packager process context. To add custom variables you can create `.env` file in the root folder of your project and add needed environment variables in the following format:
```

File: README.md [hunk 13]
```diff
@@ -404,6 +404,13 @@ If you use Android, you need to change the debug server by:
...
+
+```
+Variable1_name=Variable1_value
+Variable2_name=Variable2_value
+```
+Variables that are declared in this .env file can override the original enviroment variables from process.env of the packager.
```
Comment (SounD120) [inline README.md:413]:
```suggestion
Variables that are declared in this `.env` file can override the original environment variables from `process.env` of the Packager process.
```

File: README.md [hunk 14]
```diff
@@ -413,6 +413,8 @@ Variable2_name=Variable2_value
 ```
 Variables that are declared in this `.env` file can override the original environment variables from `process.env` of the Packager process.
 
+It is possible to transfer environment variables (`env` and `envFile`) from the `launch` or `attach` to the Packager. If these variables are defined, then they will be used, otherwise the `.env` file is used.
```
Comment (SounD120) [inline README.md:416]:
```suggestion
It is possible to transfer environment variables (`env` and `envFile`) from the `launch` or `attach` debug scenarios to the Packager. If these variables are defined, then they will be used, otherwise the `.env` file is used. 
```

File: README.md [hunk 15]
```diff
@@ -404,6 +404,16 @@ If you use Android, you need to change the debug server by:
 
 **NOTE:** Some aspects of React Native hard-code the port to the default as specified in [this issue](https://github.com/facebook/react-native/issues/9145).
 
+### Custom environment variables
+Extension supports passing custom environment variables for the React Native Packager process context. To add custom variables you can create `.env` file in the root folder of your project and add needed environment variables in the following format:
```
Comment (RedMickey) [inline README.md:408]:
```suggestion
Extension supports passing custom environment variables to the React Native Packager process context. To add custom variables you can create `.env` file in the root folder of your project and add needed environment variables in the following format:
```
